rfc/kubernetes-operator.md

```diff
@@ -0,0 +1,89 @@
+# Kubernetes operator for Spinnaker
+
+| | |
+|-|-|
+| **Status**     | **Proposed** |
+| **RFC #**      | https://github.com/spinnaker/community/pull/41 |
+| **Author(s)**  | Nicolas Cohen (`@ncknt`) |
+| **SIG / WG**   | Kubernetes SIG |
+
+## Overview
+
+Halyard is great at offering a simple way to configure Spinnaker services but is not easy to operationalize. We've come up with an MVP of a Kubernetes operator for Spinnaker that allows administrators to manage Spinnaker with kubectl (or whatever Kubernetes tool).
+
+We're proposing to move the operator to Spinnaker's GitHub: https://docs.google.com/document/d/1Fs-xK7sOZ4YLSw1ae_cQ_rsB0kMNon9AfYli2a5Udhs/edit?usp=sharing
+
+### Goals and Non-Goals
+
+The goal of the Spinnaker Operator is to enable users to manage Spinnaker via Kubernetes native tools (`kubectl`, ...). Short term we want feature parity with Halyard deployments. Longer term, we'd like extensive preflight checks and active monitoring of Spinnaker.
+
+## Motivation and Rationale
+
+Deploying Spinnaker in Kubernetes with Halyard comes with a few challenges:
+
+- Users often start using Halyard from their laptop and struggle to find a good process to share the configuration.
+- How should Halyard itself be managed? Locally spun up at CI time or as a long-time running service?
+- How do you template the configuration if you have multiple running instances?
+- How do you pass credentials to Halyard, like kubeconfig files or AWS credentials?
+- Users are left with a lot of activities that Halyard doesn’t handle such as exposing Spinnaker and cleaning up old secrets.
+
+Finally, there’s the impression that Spinnaker is not “Kubernetes native” because it cannot be installed and managed with the usual administrator toolchain (kubectl). Talking to Spinnaker administrators, we’ve realized this is a pretty widely shared feeling. They’re already proficient at using kubectl and are reluctant to learn and configure yet another new tool.
+
+## Timeline
+
+Current MVP can deploy and expose Spinnaker. Preflight checks will be added in the coming weeks. Moving the operator to Spinnaker's GitHub account involves publishing new images to gcr and cleaning the build.
+
+## Design
+
+See https://docs.google.com/document/d/1Fs-xK7sOZ4YLSw1ae_cQ_rsB0kMNon9AfYli2a5Udhs/edit#heading=h.7vexvfs9sl5e
+
+### Dependencies
+
+No new dependencies to existing projects. Some (relatively small) changes are necessary in Halyard to:
```

Comment:
I noted this on the design doc, but adding here as well---I'd like to see a more shared ownership of Halyard if we're going to be building additional dependencies on top of Halyard.  I review a large percentage of the PRs to Haylard and would like to see that responsibility shared more; I think that if this will be built on top of Hayard it makes sense for the maintainers of this operator to take a larger role in maintaining Haylard.

Reply:
That's a fair point. Thanks for putting it up with all the work. We're ready to step up and help with the load on Halyard.

Reply:
Thanks, that would be great!